src/pay/payment_consumer.py

```diff
@@ -94,11 +95,9 @@ def __init__(
     def run(self):
         running = True
         while running:
-            # Exit if disk is full
-            # https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/issues/504
             if disk_is_full():
                 running = False
-                break
+                exit_program(ExitCode.NO_SPACE)
```

Comment:
Why remove the break here and exit directly? Doesn't this change the behavior?

Reply:
well it would just leave the run at the moment and break, but if the disk is full why not just exit?